### ISSUE TITLE
Cant assign user to card? 

### ISSUE THREAD

**Issue by joomlaproffs** [nextcloud/deck#2470]
Nextcloud 20.0,0 
PHP 7.3
Deck 1.1.2

When I go to a card and try to set a user in my NextCloud install nothing happens. It adds the round icon of the user at first but if I close the card details and enter the card again, assignment are lost. Am I missing something fundamental here? The Tags input field above works fine there with the colored "Action needed", "Finsihed" etc. 

Ive tried Firefox, Chrome and Edge browser. Same behavior in all. I just cant get assignment to work at all for any card.

My install is a Unraid server with the NextCloud docker. 

Error log: 

```
[PHP] Error: Error: Undefined offset: 3 at /config/www/nextcloud/lib/private/legacy/OC_Helper.php#548 at <<closure>>

 0. /config/www/nextcloud/lib/private/legacy/OC_Helper.php line 548
    OC\Log\ErrorHandler::onError(8, "Undefined offset: 3", "/config/www/nex ... p", 548, {path: "/",rootI ... }})
 1. /config/www/nextcloud/apps/settings/lib/Settings/Personal/PersonalInfo.php line 108
    OC_Helper::getStorageInfo("/")
 2. /config/www/nextcloud/apps/settings/lib/Controller/CommonSettingsTrait.php line 141
    OCA\Settings\Settings\Personal\PersonalInfo->getForm()
 3. /config/www/nextcloud/apps/settings/lib/Controller/PersonalSettingsController.php line 77
    OCA\Settings\Controller\PersonalSettingsController->formatSettings({10: [OCA\Settin ... ]})
 4. /config/www/nextcloud/apps/settings/lib/Controller/CommonSettingsTrait.php line 152
    OCA\Settings\Controller\PersonalSettingsController->getSettings("personal-info")
 5. /config/www/nextcloud/apps/settings/lib/Controller/PersonalSettingsController.php line 68
    OCA\Settings\Controller\PersonalSettingsController->getIndexResponse("personal", "personal-info")
 6. /config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 169
    OCA\Settings\Controller\PersonalSettingsController->index("personal-info")
 7. /config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 100
    OC\AppFramework\Http\Dispatcher->executeController(OCA\Settings\Con ... {}, "index")
 8. /config/www/nextcloud/lib/private/AppFramework/App.php line 152
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Settings\Con ... {}, "index")
 9. /config/www/nextcloud/lib/private/Route/Router.php line 308
    OC\AppFramework\App::main("OCA\\Settings\\ ... r", "index", OC\AppFramework\ ... {}, {section: "perso ... "})
10. /config/www/nextcloud/lib/base.php line 1009
    OC\Route\Router->match("/settings/user")
11. /config/www/nextcloud/index.php line 37
    OC::handleRequest()

GET /settings/user
from 188.126.94.70 by claes at 2020-10-23T12:59:24+00:00
```

**Comment by jakobroehrl** [comment]
Is the board shared?
Could you provide a screenshot and steps to reproduce?

**Comment by joomlaproffs** [comment]
Yes the card is shared with a Group. I have tried all kind of different sharing - but nothing works. 

Here is a video of the problem: https://www.dropbox.com/s/g21aiycqon9zkze/2020-10-27%2010-45-11-ScreenCapture.mp4?dl=0

**Comment by adna9** [comment]
we facing exact same issue any solutions?

**Comment by stefan-niedermann** [comment]
@adna9 please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue.

**Comment by juliushaertl** [comment]
The nextcloud log error seems unrelated. Could you check the browser console for errors when picking a user for assignemnt? 

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by joomlaproffs** [comment]
I get this error in console:
![image](https://user-images.githubusercontent.com/7556099/98677426-4adf2200-235d-11eb-80b0-b9e9a7dd8b61.png)

**Comment by joomlaproffs** [comment]
I got this from my dev? Is there some tables missing? 

Object { "\u0000*\u0000message": "An exception occurred while executing 'INSERT INTO `oc_deck_assigned_users`(`card_id`,`participant`,`type`) VALUES(?,?,?)' with params [25, \"naga\", 0]:\n\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 'type' in 'field list'", "\u0000*\u0000code": 0, "\u0000*\u0000file": "/config/www/nextcloud/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php", … }

**Comment by juliushaertl** [comment]
Could you send me the output of the following command?

`occ migrations:status deck`

**Comment by joomlaproffs** [comment]
@juliushaertl 
![image](https://user-images.githubusercontent.com/7556099/98793655-7b2fca80-2408-11eb-84a4-e5077d1081f7.png)

**Comment by joomlaproffs** [comment]
I have now updated to Deck 1.2.0 via the automatic update feature in NextCloud. Should this have fixed this above issue? Or do I need to do anything more? Because my problem is still there... ?

**Comment by juliushaertl** [comment]
Could you try running the following command and watch the logs if there is any error reported?

`occ migrations:execute deck 1000Date20200308073933`

**Comment by joomlaproffs** [comment]
@juliushaertl I get "Can not go back to previous migration without debug enabled" if I do that in console. How can I enable debug?

**Comment by juliushaertl** [comment]
Ah, sorry missed that. You can temporarily enable that by setting `'debug' => true,` in your config/config.php

**Comment by joomlaproffs** [comment]
Ok after running occ migrations:execute deck 1000Date20200308073933 (no error) all seems to be working now! Thank you

**Comment by juliushaertl** [comment]
Great, I cannot really tell why this was not working in the first place on your setup, but glad that it is working now.

**Comment by Emi94** [comment]
@juliushaertl sorry to reopen a closed issue, but I have a Nextcloud instance with the same symptoms. Here's the output of `occ migrations:status deck` :
```
    >> App:                                                deck
    >> Version Table Name:                                 oc_migrations
    >> Migrations Namespace:                               OCA\Deck\Migration
    >> Migrations Directory:                               /var/www/nextcloud/apps/deck/lib/Migration
    >> Previous Version:                                   1000Date20200308073933
    >> Current Version:                                    10200Date20201111150114
    >> Next Version:                                       Already at latest migration step
    >> Latest Version:                                     10200Date20201111150114
    >> Executed Migrations:                                3
    >> Executed Unavailable Migrations:                    3
    >> Available Migrations:                               3
    >> New Migrations:                                     3
    >> Pending Migrations:                                 None
```

Should I run `occ migrations:execute deck 1000Date20200308073933` as well? Is it even safe now that there's a newer migration?

**Comment by putt1ck** [comment]
Possibly related (or possibly just another bug...), the affected instance shows disabled users as available to be assigned to cards

**Comment by putt1ck** [comment]
I can confirm that running the *specific* migrations command fixes the issue on the affected instance, even though it is showing as a later version (10200Date20201111150114)
